server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsFormat.java

```diff
@@ -0,0 +1,496 @@
+ *         <li>The procedure how PackedPayLengthBlock and PackedOffsetLengthBlock are generated is
+ *             the same as PackedFreqBlock in chapter <a href="#Frequencies">Frequencies and Skip
+ *             Data</a>. While PackedStartDeltaBlock follows a same procedure as
+ *             PackedDocDeltaBlock.
+ *         <li>PackedPayBlockNum is always equal to PackedPosBlockNum, for the same term. It is also
+ *             synonym for PackedOffsetBlockNum.
+ *         <li>SumPayLength is the total length of payloads written within one block, should be the
+ *             sum of PayLengths in one packed block.
+ *         <li>PayLength in PackedPayLengthBlock is the length of each payload associated with the
+ *             current position.
+ *       </ul>
+ * </dl>
+ *
+ * @lucene.experimental
+ */
+public final class ES812PostingsFormat extends PostingsFormat {
+
+    /**
+     * Filename extension for document number, frequencies, and skip data. See chapter: <a
+     * href="#Frequencies">Frequencies and Skip Data</a>
+     */
+    public static final String DOC_EXTENSION = "doc";
+
+    /** Filename extension for positions. See chapter: <a href="#Positions">Positions</a> */
+    public static final String POS_EXTENSION = "pos";
+
+    /**
+     * Filename extension for payloads and offsets. See chapter: <a href="#Payloads">Payloads and
+     * Offsets</a>
+     */
+    public static final String PAY_EXTENSION = "pay";
+
+    /** Size of blocks. */
+    public static final int BLOCK_SIZE = ForUtil.BLOCK_SIZE;
+
+    /**
+     * Expert: The maximum number of skip levels. Smaller values result in slightly smaller indexes,
+     * but slower skipping in big posting lists.
+     */
+    static final int MAX_SKIP_LEVELS = 10;
+
+    static final String CODEC_NAME = "ES812Postings";
+    static final String TERMS_CODEC = "ES812PostingsWriterTerms";
+    static final String DOC_CODEC = "ES812PostingsWriterDoc";
+    static final String POS_CODEC = "ES812PostingsWriterPos";
+    static final String PAY_CODEC = "ES812PostingsWriterPay";
+
+    // Increment version to change it
+    static final int VERSION_START = 0;
+    static final int VERSION_CURRENT = VERSION_START;
+
+    /** Creates read-only {@code ES812PostingsFormat}. */
+    public ES812PostingsFormat() {
+        super(CODEC_NAME);
+    }
+
+    @Override
+    public String toString() {
+        return getName();
+    }
+
+    @Override
+    public FieldsConsumer fieldsConsumer(SegmentWriteState state) throws IOException {
+        PostingsWriterBase postingsWriter = new ES812PostingsWriter(state);
+        boolean success = false;
+        try {
+            FieldsConsumer ret = new Lucene90BlockTreeTermsWriter(
+                state,
+                postingsWriter,
+                Lucene90BlockTreeTermsWriter.DEFAULT_MIN_BLOCK_SIZE,
+                Lucene90BlockTreeTermsWriter.DEFAULT_MAX_BLOCK_SIZE,
+                VERSION_START
```

Comment:
The class in Lucene does this to write data with the old format (the version number of Lucene90BlockTreeTermsWriter was bumped in 9.9). Let's use the ctor that doesn't take a version here so that it writes data with VERSION_CURRENT.

Reply:
That was one of the questions I had, thanks!